flocker/control/_persistence.py

```diff
@@ -202,7 +273,7 @@ def wire_encode(obj):
     :param obj: An object from the configuration model, e.g. ``Deployment``.
     :return bytes: Encoded object.
     """
-    return dumps(obj, cls=_ConfigurationEncoder)
+    return dumps(_cached_dfs_serialize(_cached_dfs_serialize(obj)))
```

Comment:
Why the double serialize here?

Also, would prefer (regardless of whether double-serialize is needed or not) to make this into vertical rather than horizontal:

    serialized = _cache_dfs_serialize(obj)
    really_serialized = _cache_dfs_serialize(obj) # needed because ...
    return dumps(really_serialized)